### PR TITLE
Add link to API doc to readme

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Add link to API doc to readme.

Signed-off-by: Alan Jowett <alanjo@microsoft.com>

### CODE REVIEW

File: README.md
Comment (matt-gretton-dann) [inline README.md:16]:
```suggestion
[API Documentation](https://iovisor.github.io/ubpf/ubpf_8h.html)
```

We should definitely use a `https:` link.

Would this be better to go to the top-level URL? https://iovisor.github.io/ubpf/

My concern is that the header page link looks fragile and easily broken if we change the docs.  I don't know how doxygen crates page file names and how stable it is.  Although the docs for doxygen seem to suggest that file names are stable if `SHORT_NAMES` is set to `NO` in the `Doxyfile`.  Which it is.

Happy for you to change or not as you think best.

**Review by matt-gretton-dann — CHANGES_REQUESTED** [review]
I have one comment on the URL linked to.

**Comment by Alan-Jowett** [comment]
All comments addressed. Waiting for approval.

**Pull request merged**
